ui/src/middleware.ts

```diff
@@ -1,16 +1,87 @@
-import { NextResponse } from 'next/server'
-import type { NextRequest } from 'next/server'
+import { NextResponse, NextRequest } from 'next/server'
 
 const apiPort = process.env.API_PORT || 3001
 
-export function middleware(request: NextRequest) {
-  if (request.nextUrl.pathname.startsWith('/api/')) {
-    const url = request.nextUrl.clone()
-    const destination = new URL(`http://localhost:${apiPort}`)
-    url.host = destination.host
-    url.port = destination.port
-    url.protocol = destination.protocol
-    url.basePath = ''
-    return NextResponse.rewrite(url)
+export async function middleware(request: NextRequest) {
```

Comment:
There isn't much need for this middleware. You can add an interceptor to axios so when calling any of the ApiHandlers, any 401 response would redirect the user to the login page.